src/solvers/imex_tableaus.jl

```diff
@@ -6,39 +8,91 @@ export SSP222, SSP322, SSP332, SSP333, SSP433
 
 using StaticArrays: @SArray, SMatrix, sacollect
 
-################################################################################
+"""
+    AbstractAlgorithmConstraint
+
+A mechanism for restricting which operations an algorithm is allowed to perform.
+"""
+abstract type AbstractAlgorithmConstraint end
+
+"""
+    Unconstrained
+
+Indicates that the algorithm may perform any supported operations.
+"""
+struct Unconstrained <: AbstractAlgorithmConstraint end
+
+"""
+    SSPConstrained
+
+Indicates that the algorithm must be "strong stability preserving", which
+ensures that limiters will be handled in a mathematically consistent way.
+"""
+struct SSPConstrained <: AbstractAlgorithmConstraint end
```

Comment:
I'm thinking that this should go in the `src/ClimaTimeSteppers.jl` file, at least the `abstract type AbstractAlgorithmConstraint end` definition should. Also, I think adding more description under the `AbstractAlgorithmConstraint` docs would be useful (e.g., provide an example of the SSP constraint)

Reply:
Ok, yeah, I'll move `AbstractAlgorithmConstraint` under `AbstractAlgorithmName`, and I'll improve the docs for all three structs. I was going to write something more about limiters, but I was hoping to finish the deformation flow test before saying anything for certain. I'll just write what we're currently hoping will be true.